gk/main.c

```diff
@@ -594,8 +594,9 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 	rte_hash_set_cmp_func(instance->ip_flow_hash_table, ip_flow_cmp_eq);
 
 	/* Setup the flow entry table for GK block @block_idx. */
-	instance->ip_flow_entry_table = (struct flow_entry *)rte_calloc(NULL,
-		gk_conf->flow_ht_size, sizeof(struct flow_entry), 0);
+	instance->ip_flow_entry_table = (struct flow_entry *)rte_calloc_socket(
```

Comment:
Can't we drop the cast `(struct flow_entry *)`?

Reply:
Although I haven't compiled the code, I think it should be ok to drop the cast according to many examples in DPDK (e.g., [here](https://github.com/cjdoucette/dpdk/blob/33cadd2c92883ff7d13b1a792d3a4e6b6b5c11c0/drivers/net/sfc/sfc_ef10_tx.c#L930))